numpy/linalg/setup.py

```diff
@@ -53,8 +53,41 @@ def get_lapack_lite_sources(ext, build_dir):
         extra_info=lapack_info,
         libraries=['npymath'],
     )
+
+    # TODO: link extension robustly on non-Linux
+    if len(get_info('openblas')) > 2:
+        # if openblas is being used to compile NumPy
+        # build an extension to gather more information
+        # about the openblas version used
+        config.add_extension('openblas_config',
+                             sources=['openblas_config.c'],
+                             extra_info=lapack_info,
+                             libraries=['openblas'],
+                             )
     return config
 
 if __name__ == '__main__':
     from numpy.distutils.core import setup
     setup(configuration=configuration)
+    import numpy.distutils.system_info
+    from numpy.distutils.system_info import get_info
+    if len(get_info('openblas')) > 2:
+        # openblas >= 0.3.4 provides version information;
+        # numpy/distutils/system_info has already initialized
+        # at this stage, so just update the dictionary for
+        # openblas info
+        from numpy.linalg import openblas_config
+        openblas_config_str = openblas_config._openblas_info()
+        openblas_config_list = openblas_config_str.split()
+        # slightly obscure API for system_info objects;
+        # they weren't really intended for public modification
+        # as the docs clearly indicate
+        info = system_info.system_info.saved_results['openblas_info']
+        new_info = {}
+        if openblas_config_list[0] == b"OpenBLAS":
+            # version string will be present
+            new_info['version'] = openblas_config_list[1]
+        else:
+            # older OpenBLAS config API
+            new_info['version'] = None
+        info.update(new_info)
```

Comment:
the update won't happen in time to show the openblas version in the `FOUND` block one sees when distutils runs during setup, but it would show up if called later

I suppose this makes sense maybe -- if we're associating this machinery with linalg, then the libraries have to be detected first, and then the extension code is called to expand that library info with the version number.

Reply:
In other words, we need openblas to be initially detected by distutils so that this linalg extension can be built, and then build up the openblas lib info available to the former.